assignments/assignment01.rb

```diff
@@ -68,13 +80,573 @@ def my_reverse(s)
+  array.each {|i| sum += i}
+  the_mean = sum/array.length.to_i
+  puts the_mean.to_f
+end
+
+def median(array)
+  ordered_array = array.sort
+  number_of_elements = array.length
+  if array.length % 2 != 0
+    median_pos = number_of_elements.to_i/2
+    puts ordered_array[median_pos].to_f
+  else array.length % 2 == 0
+    median1 = ordered_array[array.length.to_i/2]
+    median2 = ordered_array[array.length.to_i/2-1]
+    puts (median1.to_f+median2.to_f)/2
+  end
+end
+
+mean([1, 2, 3])
+mean([1, 1, 4])
+puts ""
+
+median([1, 2, 3])
+median([1, 1, 4])
+puts "---END OF PROBLEM 2---"
+
+def pluck(hash, target)
+  number = 0
+  if target == :name
+    number = 0
+  elsif target == :instrument
+    number = 1
+  end
+  hash.each do |i|
+    vals = i.values[number]
+    output = []
+    output<<vals
+    puts output
+  end
+end
+
+records = [{name: "John", instrument: "guitar"},
+{name: "Paul", instrument: "bass"},
+{name: "George", instrument: "guitar"},
+{name: "Ringo", instrument: "drums"}]
+
+pluck(records, :name)
+puts ""
+pluck(records, :instrument)
+puts "---END OF PROBLEM 3---"
+
+def starting_balance_method
+  starting_balance = 0.00
+end
+
+def withdrawal_transactions
+  File.open("csv.csv") do |file|
+    lines = file.readlines
+    separation_processes = lines.shift.chomp.split(",")
+    withdrawals = []
+    lines.each do |i|
+      if i.include?("withdrawal")
+        withdrawals << i
+      end
+    end
+    return withdrawals
+  end
+end
+
+def deposit_transactions
+  File.open("csv.csv") do |file|
+    lines = file.readlines
+    separation_processes = lines.shift.chomp.split(",")
+    deposits = []
+    lines.each do |i|
+      if i.include?("deposit")
+        deposits << i
+      end
+    end
+    return deposits
+  end
+end
+
+def calculate_final_balance_after_all_transactions
+  File.open("csv.csv") do |file|
+    lines = file.readlines
+    separation_processes = lines.shift.chomp.split(",")
+    withdrawals = []
+    deposits = []
+    lines.each do |i|
+      if i.include?("withdrawal")
+	withdrawals << i
+      else
+	deposits << i
+      end
+    end
+    deposit_manip = deposits.map {|i| i.split(",")}.flatten!
+    deposit_txs =  deposit_manip.reject {|i| deposit_manip.index(i).odd?}
+    deposit_sum = deposit_txs.each_slice(2).inject(0) {|i, (j,k)| i += k.to_f}
+
+    withdrawal_manip = withdrawals.map {|i| i.split(",")}.flatten!
+    withdrawal_txs = withdrawal_manip.reject {|i| withdrawal_manip.index(i).odd?}
+    withdrawal_sum = withdrawal_txs.each_slice(2).inject(0) {|i, (j,k)| i += k.to_f}
+
+    final_balance = deposit_sum - withdrawal_sum
+    return "Your balance after all transactions is $#{final_balance.round(2)}"
+  end
+end
+
+def date_to_julian(some_date)
+#method that converts the date in the format (mm/dd/yyyy) into a julian date
+#*DID NOT ACCOUNT FOR LEAP YEAR, SINCE 2014 IS NOT A LEAP YEAR*
+  date_array = some_date.split("/")
+  month = date_array[0].to_i
+  day = date_array[1].to_i
+  if month == 1
```

Comment:
this is quite brute force. I'm sure that you could have come up with a more elegant way of doing this. Here's one idea: create an array, indexed by month, with the number of days to add. In fact, you could calculate this array starting with an array that has the number of days per month.